dot/digest/digest.go

```diff
@@ -267,23 +265,29 @@ func (h *Handler) persistBABEDigestsForNextEpoch(finalizedHeader *types.Header)
 	}
 
 	nextEpoch := currEpoch + 1
-	err = h.epochState.FinalizeBABENextEpochData(nextEpoch)
-	if err != nil && !errors.Is(err, state.ErrEpochNotInMemory) {
-		return fmt.Errorf("cannot finalize babe next epoch data for block number %d (%s): %w",
-			finalizedHeader.Number, finalizedHeader.Hash(), err)
+
+	appliedEpochData, appliedConfigData, err := h.epochState.AlreadyDefined(nextEpoch)
+	if err != nil {
+		return fmt.Errorf("cannot check if next epoch is already defined: %w", err)
 	}
 
-	err = h.epochState.FinalizeBABENextConfigData(nextEpoch)
-	if err == nil {
-		return nil
-	} else if errors.Is(err, state.ErrEpochNotInMemory) {
-		return fmt.Errorf("%w: %s", ErrDefineNextEpoch, err)
+	if !appliedEpochData {
+		err = h.epochState.FinalizeBABENextEpochData(nextEpoch)
+		if err != nil && !errors.Is(err, state.ErrEpochNotInMemory) {
+			return fmt.Errorf("cannot finalize babe next epoch data for block number %d (%s): %w",
+				finalizedHeader.Number, finalizedHeader.Hash(), err)
+		}
 	}
 
-	// the epoch state does not contains any information about the next epoch
-	return fmt.Errorf("cannot finalize babe next config data for block number %d (%s): %w",
-		finalizedHeader.Number, finalizedHeader.Hash(), err)
+	if !appliedConfigData {
+		err = h.epochState.FinalizeBABENextConfigData(nextEpoch)
+		if err != nil && !errors.Is(err, state.ErrEpochNotInMemory) {
+			return fmt.Errorf("cannot finalize babe next config data for block number %d (%s): %w",
+				finalizedHeader.Number, finalizedHeader.Hash(), err)
+		}
```

Comment:
not every epoch will have `ConfigData`, if it doesn't, it should just use the last set config data. does this handle that case? it seems to me  like it doesn't